based_build/localization_helper.py

```diff
@@ -0,0 +1,192 @@
+# pyright: reportImplicitOverride=false, reportUninitializedInstanceVariable=false
+
+from __future__ import annotations
+
+import json
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, ClassVar, Dict, TypedDict
+
+from rich.highlighter import ReprHighlighter
+from rich.text import Text
+from textual import work
+from textual.app import App, ComposeResult
+from textual.color import Color
+from textual.screen import Screen
+from textual.widgets import Footer, Header, Label, Tabs, Tree
+
+if TYPE_CHECKING:
+    from textual.widgets.tree import TreeNode
+
+highlighter = ReprHighlighter()
+
+LOCFILES_BASE = (
+    Path(__file__).parent.parent / "packages" / "pyright-internal" / "src" / "localization"
+)
+
+ALL_LANGUAGES = [t for x in LOCFILES_BASE.iterdir() if (t := x.stem[12:])]
+
+
+class _LocMsgComment(TypedDict):
+    message: str
+    comment: str
+
+
+LocMessages = Dict[str, "str | _LocMsgComment"]
+
+
+def get_locfile(language: str = "en-us"):
+    return LOCFILES_BASE / f"package.nls.{language}.json"
+
+
+with get_locfile().open() as f:
+    LOCDATA_EN_US: Dict[str, LocMessages] = json.load(f)  # noqa: UP006
+
+
+def diff_keys(orig: dict[str, Any], comp: dict[str, Any]):
+    msgs: list[str] = []
+    orig_set = set(orig.keys())
+    comp_set = set(comp.keys())
+    if orig_set != comp_set:
+        msgs.append("Compared to the original:")
+        if missing := orig_set - comp_set:
+            msgs.append("these keys are missing: " + ", ".join(missing) + ".")
+        if unused := comp_set - orig_set:
+            msgs.append("these keys are unused: " + ", ".join(unused) + ".")
+    else:
+        msgs.append("This object has the same keys as the original.")
+    return "\n".join(msgs)
+
+
+class LocDataTree(Tree[str]):
+    _locnode: ClassVar[Dict[str, TreeNode[str]]] = {}  # noqa: UP006
```

Comment:
oh right, forgot to mention that this isn't *always* true. for example with type aliases they still get evaluated at runtime regardless because they're just a normal variable assignment. i fixed it in 5760a018f1b2e90be27480814ae7bdf061b4201c